### ISSUE TITLE
7.1.0 version causing the issue undefined method `deprecator' for ActiveSupport:Module (NoMethodError)

### ISSUE THREAD

**Issue by ykhandelwal913** [rails/rails#49495]
### Steps to reproduce

running pod install which internally calls ruby.
`pod install --repo-update`
```
Traceback (most recent call last):
	11: from /Users/xxxx/.rvm/gems/ruby-2.7.2/bin/ruby_executable_hooks:22:in `<main>'
	10: from /Users/xxxx/.rvm/gems/ruby-2.7.2/bin/ruby_executable_hooks:22:in `eval'
	 9: from /Users/xxxx/.rvm/gems/ruby-2.7.2/bin/pod:25:in `<main>'
	 8: from /Users/xxxx/.rvm/gems/ruby-2.7.2/bin/pod:25:in `load'
	 7: from /Users/xxxx/.rvm/gems/ruby-2.7.2/gems/cocoapods-1.13.0/bin/pod:36:in `<top (required)>'
	 6: from /Users/xxxx/.rvm/rubies/ruby-2.7.2/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:85:in `require'
	 5: from /Users/xxxx/.rvm/rubies/ruby-2.7.2/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:85:in `require'
	 4: from /Users/xxxx/.rvm/gems/ruby-2.7.2/gems/cocoapods-1.13.0/lib/cocoapods.rb:9:in `<top (required)>'
	 3: from /Users/xxxx/.rvm/rubies/ruby-2.7.2/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:85:in `require'
	 2: from /Users/xxxx/.rvm/rubies/ruby-2.7.2/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:85:in `require'
	 1: from /Users/xxxx/.rvm/gems/ruby-2.7.2/gems/activesupport-7.1.0/lib/active_support/core_ext/array/conversions.rb:8:in `<top (required)>'
/Users/xxxx/.rvm/gems/ruby-2.7.2/gems/activesupport-7.1.0/lib/active_support/core_ext/array/conversions.rb:108:in `<class:Array>': undefined method `deprecator' for ActiveSupport:Module (NoMethodError)
Did you mean?  deprecate_constant`
```

### System configuration
**Rails version**:

**Ruby version**:
2.7.2

**Comment by hromovp** [comment]
seeing this as well

**Comment by efstathiosntonas** [comment]
same here

**Comment by fatkodima** [comment]
Can you guys provide steps to reproduce?

**Comment by efstathiosntonas** [comment]
@fatkodima I just started seeing this on our CI, for some reason it still works locally. After `gem install cocoapods` we can no longer install with `pod install`

**Comment by hromovp** [comment]
> @fatkodima I just started seeing this on our CI, for some reason it still works locally. After `gem install cocoapods` we can no longer install with `pod install`

Same for me, I think this should work on bare react native project with the latest gems

**Comment by ykhandelwal913** [comment]
Downgrading the active support to 7.0.8 worked

**Comment by efstathiosntonas** [comment]
@ykhandelwal913 how to downgrade it? I have minimum knowledge around gems 🤷

**Comment by fatkodima** [comment]
So the problem is this line https://github.com/CocoaPods/CocoaPods/blob/479e71131962e3e560e4cc0601d39860c2a590ce/lib/cocoapods.rb#L9
requires a file, which then executes https://github.com/rails/rails/blob/2d271a4b901e7a7dae8ba4e5e83edfebdee278e8/activesupport/lib/active_support/core_ext/array/conversions.rb#L108, but `deprecator` is defined in https://github.com/rails/rails/blob/main/activesupport/lib/active_support/deprecator.rb which is required by https://github.com/rails/rails/blob/2d271a4b901e7a7dae8ba4e5e83edfebdee278e8/activesupport/lib/active_support.rb#L29

So, I think the solution is to add `require "activesupport"` right before these lines https://github.com/CocoaPods/CocoaPods/blob/479e71131962e3e560e4cc0601d39860c2a590ce/lib/cocoapods.rb#L7-L9 in the `cocoapods` itself.

**Comment by davidrunger** [comment]
I am seeing this exception not while using `cocoapods`, but rather simply from requiring `active_support/testing/time_helpers` (which I do in my `spec_helper.rb` for a project using RSpec).

Here's a very simple reproduction:

```rb
require 'bundler/inline'

gemfile(true) do
  source 'https://rubygems.org'

  gem 'activesupport', '7.1.0'
end

require 'active_support/testing/time_helpers'
```

Result:

```
❯ ruby reproduction.rb
Fetching gem metadata from https://rubygems.org/........
Resolving dependencies...
Using base64 0.1.1
[... omitted similar lines ...]
Using activesupport 7.1.0
Using bundler 2.4.13
/Users/david/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.1.0/lib/active_support/core_ext/array/conversions.rb:108:in `<class:Array>': undefined method `deprecator' for ActiveSupport:Module (NoMethodError)

  deprecate to_default_s: :to_s, deprecator: ActiveSupport.deprecator
                                                          ^^^^^^^^^^^
Did you mean?  deprecate_constant
	from /Users/david/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.1.0/lib/active_support/core_ext/array/conversions.rb:8:in `<top (required)>'
	from <internal:/Users/david/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:38:in `require'
	from <internal:/Users/david/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:38:in `require'
	from /Users/david/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.1.0/lib/active_support/duration.rb:3:in `<top (required)>'
	from <internal:/Users/david/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:38:in `require'
	from <internal:/Users/david/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:38:in `require'
	from /Users/david/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.1.0/lib/active_support/core_ext/time/calculations.rb:3:in `<top (required)>'
	from <internal:/Users/david/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:38:in `require'
	from <internal:/Users/david/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:38:in `require'
	from /Users/david/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.1.0/lib/active_support/testing/time_helpers.rb:4:in `<top (required)>'
	from <internal:/Users/david/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:38:in `require'
	from <internal:/Users/david/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:38:in `require'
	from reproduction.rb:9:in `<main>'
```

**Comment by fatkodima** [comment]
You also need to add `require "active_support"` before that.  See https://guides.rubyonrails.org/active_support_core_extensions.html#cherry-picking-a-definition

**Comment by ykhandelwal913** [comment]
gem uninstall -v “7.1.0”
gem install -v “7.0.8”

this is just a workaround until the cocoapod team fix the issue

**Comment by josegbel** [comment]
Also getting this 👀

**Comment by TheFabbiusCorp** [comment]
> gem uninstall -v “7.1.0” gem install -v “7.0.8”
> 
> this is just a workaround until the cocoapod team fix the issue

This is not sufficient if you are running your builds in a pipeline where the latest pods get fetched every new run.

If you use a Gemfile you need to add this line to it:
`gem "activesupport", "~> 7.0.8"`

**Comment by rafaelfranca** [comment]
If you are not in a Rails application (that already does that for you automatically), you should always require `active_support` beforing requiring anything else inside `active_support`. This is a requirement from the framework since the beginning.

See https://guides.rubyonrails.org/active_support_core_extensions.html#stand-alone-active-support

This is an issue in CocoaPods, or any gems that requires Active Support features before requiring the `active_support` entrypoint.

**Comment by ghiculescu** [comment]
Looks like it is fixed by https://github.com/CocoaPods/CocoaPods/pull/12082

**Comment by llaichiyu** [comment]
> gem install -v “7.0.8”

thanks a lot ,i have met this question,and i 'v already use  '**gem install  activatesupport -v “7.0.8”** '，and i can use pod --version.

**Comment by llaichiyu** [comment]
> > gem install -v “7.0.8”
> 
> thanks a lot ,i have met this question,and i 'v already use '**gem install activatesupport -v “7.0.8”** '，and i can use pod --version.

<img width="558" alt="image" src="https://github.com/rails/rails/assets/44918169/138dd808-7cee-40e4-9580-b7b5ad82e920">

**Comment by llaichiyu** [comment]
> llaichiyu

i have made a mistake which i used wrong package name.
the wrong is **activatesupport** and the right is **activesupport**

![image](https://github.com/rails/rails/assets/44918169/7ddb1b88-627f-4d8a-89cc-3dfad0ec6444)

**Comment by konguele** [comment]
Hello,

When I try the workaround I receive the following error. Could you please help me?:

sudo gem uninstall -v “7.1.0”
Password:
ERROR:  While executing gem ... (Gem::Requirement::BadRequirementError)
    Illformed requirement ["“7.1.0”"]

      raise BadRequirementError, "Illformed requirement [#{obj.inspect}]"
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/requirement.rb:107:in `parse'
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/requirement.rb:139:in `block in initialize'
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/requirement.rb:139:in `map!'
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/requirement.rb:139:in `initialize'
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/version_option.rb:56:in `new'
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/version_option.rb:56:in `block in add_version_option'
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/optparse/lib/optparse.rb:583:in `conv_arg'
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/optparse/lib/optparse.rb:738:in `parse'
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/optparse/lib/optparse.rb:1686:in `block in parse_in_order'
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/optparse/lib/optparse.rb:1636:in `catch'
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/optparse/lib/optparse.rb:1636:in `parse_in_order'
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/optparse/lib/optparse.rb:1630:in `order!'
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/optparse/lib/optparse.rb:1739:in `permute!'
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/optparse/lib/optparse.rb:1764:in `parse!'
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/command.rb:446:in `handle_options'
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/command.rb:314:in `invoke_with_build_args'
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/command_manager.rb:253:in `invoke_command'
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/command_manager.rb:193:in `process_args'
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/command_manager.rb:151:in `run'
	/usr/local/lib/ruby/site_ruby/3.2.0/rubygems/gem_runner.rb:52:in `run'
	/usr/local/bin/gem:12:in `<main>'

**Comment by faizy-ahmed** [comment]
Delete `Gemfile.lock` and `run bundler exec pod install`
It resolves my issue on cocoapods 1.13.0.
